client/data/marketplace/utils.ts

```diff
@@ -11,14 +11,10 @@ export const getPluginsListKey = (
 		options.searchTerm || '',
 		options.pageSize || '',
 		options.locale || '',
+		options.page || '',
+		infinite ? 'infinite' : '',
+		options.tag && ! options.searchTerm ? options.tag : '',
 	];
-	if ( infinite ) {
-		keyParams.push( 'infinite' );
-	} else {
-		keyParams.push( options.page || '' );
-	}
-	if ( options.tag && ! options.searchTerm ) {
-		keyParams.push( options.tag );
-	}
+
 	return [ key, ...keyParams ];
 };
```

Comment:
@gcsecsey are you ok with the changes in here? I'm ensuring options always reside in the same key position by no longer making things optionally pushed. It wasn't actually required for this PR but was something I was debugging and felt it was worth making more specific. Do you agree?

There is a subtle difference in that page and infinite are always included regardless. I couldn't really test this because we don't have anything that sends page values. 

This might highlight:

```
	if ( infinite ) {
		keyParams.push( 'infinite' );
	}
	if ( options.page ) {
		keyParams.push( options.page );
	}

```

Reply:
I agree it's worth making it more specific. Based on [how query keys work](https://tkdodo.eu/blog/effective-react-query-keys), the main point with this function is to produce the same array for the same set of parameters.

I think the `page` param is a leftover from before we've added the [InfiniteQuery hook](https://github.com/Automattic/wp-calypso/pull/62519/files#diff-4aa4ee82e625627654fb119928b8c515c079207f6273d88cb27bbf02d1fc3feeR80). AFAIK we don't use paginated queries anywhere else anymore, as InfiniteQuery handles the `pageParam` in its callback.

I've searched around a bit, and it seems that these can now be removed, along with [their stores](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/plugins/wporg/actions.js#L29). AFAIK these aren't used since the react-query implementation. What do you think?

Reply:
@gcsecsey can we add a ticket to do a cleanup on plugins state?

Reply:
I'd like to leave pagination around for a bit. Eventually we're going to want to show plugins in a logged out view. I'm wondering if we might opt for pagination on that view for seo purposes.